encoding/thrift/inbound.go

```diff
@@ -57,7 +58,9 @@ func (t thriftUnaryHandler) Handle(ctx context.Context, treq *transport.Request,
 		return err
 	}
 
-	body, err := ioutil.ReadAll(treq.Body)
+	buf := buffer.Get()
+	_, err := io.Copy(buf, treq.Body)
+	defer buffer.Put(buf)
```

Comment:
Nit: inline the error into the if statement and defer the buffer.Put right after getting it.

```
buf := buffer.Get()
defer buffer.Put(buf)
if _, err := io.Copy(buf, treq.Body); err != nil {
  return err
}
```